### ISSUE TITLE
[IDEA] Publish TiddlyWiki.com edition with Local Storage plugin

### ISSUE THREAD

**Issue by AnthonyMuscio** [Jermolene/TiddlyWiki5#7344]
**Is your feature request related to a problem? Please describe.**
It is common to use tiddlywiki.com as test environment alongside document repository. It has a lot of content which helps in many test cases. If you make changes you must use the download saver to keep the changes, and start managing the file(s) localy. I have proposed the inclusion of the Local Storage plugin on tiddlywiki.com so changes can persist across browser sessions, even if in an inactive state requiring the checkbox "Use browser local storage" to be ticked.
* It is understandable the overhead may need to be avoided from the key website so I propose the following;

**Describe the solution you'd like**
Just as there are a number of [Plugin demo sites](https://tiddlywiki.com/#Plugin%20Editions) could we also have one for the Browser Storage Plugin, but also a version the full tiddlywiki documentation content. This can be built such that updates to tiddlywiki.com also flow through to this "copy".
* A simple published empty.html version with browser storage may also be useful.

**Describe alternatives you've considered**
Aside from maintaining a tiddlyhost site and needing to update regularly the only option is to download tiddlywiki.com and install the browser storage plugin. This is not a shared community solution.

**Additional context**
I can explain the value of a scratch site that can save changes further if it is not self evident.
I would most likely have a bookmarklet prepared to configure such a site for use as I would like it.

**Comment by AnthonyMuscio** [comment]
Post Script
There would also be value doing this for pre-release versions as well, that is including the Browser storage plugin.

**Comment by pmario** [comment]
Related to: 

-  [IDEA] I think, the Browser Local Storage plugin needs some love #5745 
-  local Storage Plugin installed on tiddlywiki.com #4294 
-  Local Storage on 5.1.20 #3945 

Slightly related to:

-  [tw.com] Front page redesign proposal #3925 
   - https://github.com/Jermolene/TiddlyWiki5/issues/3925#issuecomment-484956876
   - >I agree that there's quite a schism between the needs of new users vs. experienced users. I wonder if we might (a) introduce a top menu bar with standard items like Learn, Docs, Community, Download and (b) use the new local storage plugin to make the current setting sticky

**Comment by saqimtiaz** [comment]
> it needs to be set up such that users need to explicitly opt-in before browser storage is used.

It may be prudent to go a step further and display a message while the browser storage is in use, to inform users that changes are being temporarily saved to their browser and do no impact what others see.

**Comment by Jermolene** [comment]
> It may be prudent to go a step further and display a message while the browser storage is in use, to inform users that changes are being temporarily saved to their browser and do no impact what others see.

Yes, that's a good idea, and probably useful in many other scenarios.

**Comment by pmario** [comment]
Yea, That's a strange thing, that most user are so used to "data collection" that they expect it. They cannot think about a mechanism we use, where no servers are involved other than serving the file. ... 

I wanted to create a proof of concept with my github test wiki. 

One of my concerns als is the [GDPR](https://en.wikipedia.org/wiki/General_Data_Protection_Regulation), which does _not_ apply for TW. ... BUT we need some info that we do not collect any user related info. Just to make that clear. ... 

I wanted to use an "Enable local saving functionality" button, which enables local-storage. With a link to a bit more info about what's going on. 

The default should be "save everything" so eg: Menu-bar config would be stored when I return to tiddlywiki.com ...

**Comment by pmario** [comment]
Since TW is served with GitHub pages we may need a link to: https://docs.github.com/en/site-policy/privacy-policies/github-privacy-statement#your-choices-regarding-our-processing-of-your-personal-data anyway. 

I don't know, what exactly GH collects if they serve GH pages. We may have to have a closer look

@Jermolene ... I think there is no such info yet?

**Comment by AnthonyMuscio** [comment]
#3945 3945 addresses this as well. [see my comment](https://github.com/Jermolene/TiddlyWiki5/issues/3945#issuecomment-1555407108) if this is acceptable and actioned a number of these threads can be closed

**Comment by Jermolene** [comment]
Hi @AnthonyMuscio if I understand correctly, you are suggesting modifying the browser storage plugin so that it can be used in a mode where it is inactive by default and if switched on by visitors the setting would persist.

I think that is already possible. Try these steps:

1. Visit https://tiddlywiki.com/editions/full/
2. Create a new tiddler and then refresh the page to verify that changes are not being saved by default
3. Open the "Browser Storage" plugin, switch to the "settings" tab
4. Locate the text box containing `[prefix[$:/state/]] -[prefix[$:/state/popup/]]`, and edit it to `[all[]]`
5. Create and save a new tiddler for testing purposes
6. Refresh the browser, and verify that the tiddler is still there

It would be easy to make the process for enabling browser storage be a single button click.

On a related side note, I am a little disappointed that only a very few people use the ["Share" edition of TiddlyWiki](https://tiddlywiki.com/share) to share demos and prototypes on the discussion forum. It works really well now, and is much, much easier than downloading and importing a JSON file. So, perhaps we should add the browser storage plugin to the "Share" edition in the first instance, to encourage its use.

**Comment by AnthonyMuscio** [comment]
Yes the full edition permits this use. I just thought and empty + local storage as we have for other plugins would allow a minimalist start. I will look at adding something to the doco.

But I also see value with a full tiddlywiki.com + local storage to allow self documentation and annotation even if its a seperate wiki.
